ios/brave-ios/Sources/Brave/WebFilters/CustomFilterListStorage.swift

```diff
@@ -148,37 +154,31 @@ import WebKit
     }
 
     let fileURL = try customRulesFileURL()
-    if FileManager.default.fileExists(atPath: fileURL.path) {
-      try FileManager.default.removeItem(at: fileURL)
+    if await AsyncFileManager.default.fileExists(atPath: fileURL.path) {
+      try await AsyncFileManager.default.removeItem(at: fileURL)
     }
 
     try customRules.write(to: fileURL, atomically: true, encoding: .utf8)
     Self.customRuleListVersion.value += 1
-    guard let fileInfo = try savedCustomRulesFileInfo() else { return }
+    guard let fileInfo = try await savedCustomRulesFileInfo() else { return }
     await AdBlockGroupsManager.shared.updateImmediately(fileInfo: fileInfo)
   }
 
   /// Delete the saved custom filter list rules
   func deleteCustomRules() async throws {
     guard let fileURL = try savedCustomRulesFileURL() else { return }
-    try FileManager.default.removeItem(at: fileURL)
+    try await AsyncFileManager.default.removeItem(at: fileURL)
     await AdBlockGroupsManager.shared.removeFileInfoImmediately(for: .filterListText)
   }
 
   /// Get or create a cache folder for the given `Resource`
   ///
   /// - Note: This technically can't really return nil as the location and folder are hard coded
-  private func getOrCreateCustomRulesFolder() throws -> URL {
-    guard
-      let folderURL = FileManager.default.getOrCreateFolder(
-        name: "custom_rules",
-        location: .applicationSupportDirectory
-      )
-    else {
-      throw ResourceFileError.failedToCreateCacheFolder
-    }
-
-    return folderURL
+  private func getOrCreateCustomRulesFolder() async throws -> URL {
+    try await AsyncFileManager.default.url(
+      for: .applicationSupportDirectory,
+      appending: "custom_rules"
+    )
```

Comment:
It's a little confusing if this creates the folder or not. Why is `create: true` not being passed? This folder needs to be created if it doesn't exist otherwise filter lists won't work on a fresh install.

Reply:
Ahh I see there is an extension which sets a default `create: Bool`

Reply:
Yes this method is the replacement of `getOrCreateFolder`